payments-core/src/test/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncherViewModelTest.kt

```diff
@@ -124,6 +150,93 @@ class GooglePayPaymentMethodLauncherViewModelTest {
             )
     }
 
+    @Test
+    fun `Factory gets initialized by Injector when Injector is available`() {
+        scenario.onFragment { fragment ->
+            val mockBuilder = mock<GooglePayPaymentMethodLauncherViewModelSubcomponent.Builder>()
+            val mockSubcomponent = mock<GooglePayPaymentMethodLauncherViewModelSubcomponent>()
+
+            whenever(mockBuilder.build()).thenReturn(mockSubcomponent)
+            whenever(mockBuilder.savedStateHandle(any())).thenReturn(mockBuilder)
+            whenever(mockBuilder.args(any())).thenReturn(mockBuilder)
+            whenever(mockSubcomponent.viewModel).thenReturn(viewModel)
+
+            val injector = object : Injector {
+                override fun inject(injectable: Injectable<*>) {
+                    val factory = injectable as GooglePayPaymentMethodLauncherViewModel.Factory
+                    factory.subComponentBuilder = mockBuilder
+                }
+            }
+            val injectorKey = 1
+            WeakMapInjectorRegistry.register(injector, injectorKey)
+            val factory = GooglePayPaymentMethodLauncherViewModel.Factory(
+                ApplicationProvider.getApplicationContext(),
+                GooglePayPaymentMethodLauncherContract.Args(
+                    mock(),
+                    "usd",
+                    1099,
+                    null,
+                    GooglePayPaymentMethodLauncherContract.Args.InjectionParams(
+                        injectorKey,
+                        emptySet(),
+                        false,
+                        "key",
+                        null
+                    )
+                ),
+                fragment
```

Comment:
Nice, didn't realize you can do this 👍

Reply:
It breaks if I pass a mock here, it needs an actual Activity or Fragment